server/tests/audit_math/test_supersimple_raire.py

```diff
@@ -0,0 +1,1146 @@
+    assert discrepancies[60000]["weighted_error"] == Decimal(-2) / Decimal(20000)
+
+
+def test_two_vote_overstatement_discrepancies(cvrs, assertions):
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest A": {
+                    "winner": 2,
+                    "loser": 1,
+                },  # One of the reported winner ballots was actually a loser ballot
+            },
+        }
+    }
+
+    contest = "Contest A"
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions[contest][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 2
+    assert discrepancies[0]["weighted_error"] == Decimal(2) / Decimal(20000)
+
+
+def test_race_not_in_cvr_discrepancy(cvrs, assertions):
+
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest F": {
+                    "winner": 0,
+                    "loser": 1,
+                },  # The audit board found a race not in the CVR
+            },
+        }
+    }
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest F"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 1
+    # Again, this should be impossible.
+    assert discrepancies[0]["weighted_error"] == Decimal("inf")
+
+
+def test_race_not_in_sample_discrepancy(cvrs, assertions):
+
+    sample_cvr = {
+        0: {
+            "times_sampled": 1,
+            "cvr": {
+                "Contest A": {"winner": 0, "loser": 0},
+                "Contest B": {"winner": 1, "loser": 2},
+                "Contest C": {"winner": 1, "loser": 2},
+                "Contest E": {"winner": 1, "loser": 2},
+            },
+        }
+    }
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest D"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 1
+    assert discrepancies[0]["weighted_error"] == Decimal(1) / Decimal(2000)
+
+
+def test_ballot_not_found_discrepancy(cvrs, assertions):
+    sample_cvr = {0: {"times_sampled": 1, "cvr": None}}
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest D"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 2
+    assert discrepancies[0]["weighted_error"] == Decimal(2) / Decimal(2000)
+
+
+def test_ballot_not_in_cvr(assertions):
+    cvrs = {}
+    sample_cvr = {
+        0: {"times_sampled": 1, "cvr": {"Contest D": {"winner": 1, "loser": 2}}}
+    }
+
+    discrepancies = supersimple_raire.compute_discrepancies(
+        cvrs, sample_cvr, assertions["Contest D"][0]
+    )
+
+    assert discrepancies
+    assert discrepancies[0]["counted_as"] == 2
+    # This should actually be impossible, unless we're already doing a recount.
```

Comment:
I would think this is possible if a ballot is in the manifest but not the CVR for some reason

Reply:
Yeah, you're right. I will fix.